packages/flutter_migrate/lib/src/environment.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:process/process.dart';
+
+import 'base/common.dart';
+import 'base/logger.dart';
+
+/// Polls the flutter tool for details about the environment and project and exposes it as
+/// a mapping of String keys to values.
+///
+/// This class is based on the `flutter analyze --suggestions --machine` flutter_tools command
+/// which dumps various variables as JSON.
+class FlutterToolsEnvironment {
+  FlutterToolsEnvironment({
+    required Map<String, Object?> mapping,
+  }) : _mapping = mapping;
+
+  /// Creates a FlutterToolsEnvironment instance by calling `flutter analyze --suggestions --machine`
+  /// and parsing its output.
+  static Future<FlutterToolsEnvironment> initializeFlutterToolsEnvironment(
+      ProcessManager processManager, Logger logger) async {
+    final ProcessResult result = await processManager
+        .run(<String>['flutter', 'analyze', '--suggestions', '--machine']);
+    if (result.exitCode != 0) {
+      if ((result.stderr as String).contains(
+          'The "--machine" flag is only valid with the "--version" flag.')) {
+        logger.printError(
+            'The migrate tool is only compatible with flutter tools 3.4.0 or newer (git hash: 21861423f25ad03c2fdb33854b53f195bc117cb3).');
+      }
+      throwToolExit(
+          'Flutter tool exited while running `flutter analyze --suggestions --machine` with: ${result.stderr}');
+    }
+    String commandOutput = result.stdout;
+    Map<String, Object?> mapping = <String, Object?>{};
+    if (commandOutput.contains('{') && commandOutput.endsWith('}\n')) {
```

Comment:
Why is the end check so specific here? It seems like it would be a lot less fragile to minor whitespace changes to either use a regex, or to `trim()` the output and check for `}` at the end.